src/components/dialogs/parameters/dynamicsimulation/curve/common/checkbox-treeview.tsx

```diff
@@ -0,0 +1,238 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import * as React from 'react';
+import { forwardRef, useCallback, useImperativeHandle, useMemo, useState } from 'react';
+import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
+import ChevronRightIcon from '@mui/icons-material/ChevronRight';
+import { alpha, Checkbox, SxProps, Theme, useTheme } from '@mui/material';
+import { styled } from '@mui/material/styles';
+import { treeItemClasses, TreeView, TreeItem } from '@mui/x-tree-view';
+import { ModelVariable } from '../dialog/model-filter';
+
+export enum CheckState {
+    UNCHECKED = 'UNCHECKED',
+    CHECKED = 'CHECKED',
+    INDETERMINATE = 'INDETERMINATE',
+}
+
+const BorderedTreeItem = styled(TreeItem)(({ root }: { root: boolean }) => {
+    const theme = useTheme();
+    const border = `1px dashed ${alpha(theme.palette.text.primary, 0.4)}`;
+    return {
+        position: 'relative',
+        '&:before': {
+            pointerEvents: 'none',
+            content: '""',
+            position: 'absolute',
+            width: 16,
+            left: -16,
+            top: 20,
+            borderBottom: root ? 'none' : border,
+        },
+        [`& .${treeItemClasses.group}`]: {
+            marginLeft: 15,
+            paddingLeft: 18,
+            borderLeft: border,
+        },
+        [`& .${treeItemClasses.label}`]: {
+            whiteSpace: 'nowrap',
+        },
+    };
+});
+
+export interface GetSelectedItemsHandle {
+    api: {
+        getSelectedItems: () => ModelVariable[];
+    };
+}
+
+interface CheckBoxTreeViewProps {
+    data: ModelVariable[];
```

Comment:
It is so specific with ModalVariable, I have a proposition here, a PR with some others recommendation: https://github.com/gridsuite/gridstudy-app/pull/2418